lib/fake_agent.rb

```diff
@@ -9,10 +9,14 @@ def initialize(endpoint = DEFAULT_ENDPOINT)
     @endpoint = endpoint
   end
 
-  def register
-    response = Typhoeus.post(registration_url, body: { agent: { hostname: hostname } })
-    json = JSON.parse(response.body)
-    @id = json["id"]
+  def register(id = ENV['AGENT_ID'])
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.